app/models/street/StreetEdgeTable.scala

```diff
@@ -139,17 +254,50 @@ object StreetEdgeTable {
     * @param auditCount
     * @return
     */
-  def auditedStreetDistance(auditCount: Int): Float = db.withSession { implicit session =>
+  def auditedStreetDistance(auditCount: Int, auditTaskQuery: Query[AuditTaskTable, AuditTask, Seq] = completedAuditTasks): Float = db.withSession { implicit session =>
     // DISTINCT query: http://stackoverflow.com/questions/18256768/select-distinct-in-scala-slick
     val edges = for {
-      (_streetEdges, _auditTasks) <- streetEdgesWithoutDeleted.innerJoin(completedAuditTasks).on(_.streetEdgeId === _.streetEdgeId)
+      (_streetEdges, _auditTasks) <- streetEdgesWithoutDeleted.innerJoin(auditTaskQuery).on(_.streetEdgeId === _.streetEdgeId)
     } yield _streetEdges
 
+    // TODO: Audit Count is not taken into account. Currently it's calculated on all completed audit tasks
```

Comment:
Need change here.

Reply:
Yeah you are correct.

Reply:
@misaugstad I need your slick expertise in doing this.